src/util/simplify_expr_pointer.cpp

```diff
@@ -297,8 +296,7 @@ simplify_exprt::simplify_pointer_offset(const unary_exprt &expr)
       {
         // (T *)0x1234 -> 0x1234
         exprt tmp = typecast_exprt(ptr.op0(), expr.type());
-        simplify_node(tmp);
-        return tmp;
+        return changed(simplify_node(tmp));
```

Comment:
Actually applies earlier as well: shouldn't we be using `simplify_typecast` here?

Reply:
Yes, probably sufficient.